appng-core/src/main/java/org/appng/core/controller/SessionListener.java

```diff
@@ -205,6 +206,21 @@ protected void setSecureFlag(HttpServletRequest httpServletRequest, Site site) {
 
 	public void requestDestroyed(ServletRequestEvent sre) {
 		MDC.clear();
+		DefaultEnvironment env = DefaultEnvironment.get(sre.getServletContext());
```

Comment:
Would be sufficient to do this in line 214, once we're sure the `env` is needed....